src/librustc_mir/transform/simplify_cfg.rs

```diff
@@ -126,7 +105,6 @@ impl<'tcx> MirPass<'tcx> for SimplifyCfg {
         while changed {
             changed = self.simplify_branches(mir);
             changed |= self.remove_goto_chains(mir);
-            self.remove_dead_blocks(mir);
```

Comment:
Also, it seems like this might take longer to reach a fixed point, since in some cases (e.g. simplify_branches) you may now have to do one step to modify the terminator and another round to remove the block, and then a final round, whereas before I think you would have finished in two?